tests/forking_agent/test_venv_manager.py

```diff
@@ -187,3 +192,178 @@ async def test_environment_creation_locking(pip_index, tmpdir) -> None:
     await venv_manager_2.start()
     assert manager.pool.keys() == venv_manager_2.pool.keys()
     await venv_manager_2.request_shutdown()
+
+
+@pytest.mark.parametrize("auto_start_agent", [True])
+async def test_remove_executor_virtual_envs(
+    clienthelper,
```

Comment:
I find this hard to test properly. Do you think this test is sufficient? Do you see says to test this better?

Reply:
What I think would be valuable, if we can achieve it, is a test that verifies that ongoing deploys are allowed to finish, and no new deploys are started.

I believe a good place to do this could be `tests/deploy/test_scheduler_agent`. It has some helpers that allow direct control over "deploys", so that you could start one, then call the endpoint to remove all venvs, then verify that the deploy is still ongoing, trigger a new one, verify that it doesn't start, trigger the old one to finish, verify that venvs get cleaned up and that the new deploy starts afterwards.

This one here would be the e2e test then, but less thoroughly. The one I suggest would be the thorough "unit" test (no server, just the agent with a mock scheduler and executor manager).